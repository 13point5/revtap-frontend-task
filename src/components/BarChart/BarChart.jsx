import React from "react";
import PropTypes from "prop-types";

import {
  BarChart,
  Bar,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
} from "recharts";
import { Card, Empty } from "antd";

function BarChartView({ data, ...restProps }) {
  return (
    <Card title="Orders Count" {...restProps}>
      {data.length > 0 ? (
        <BarChart data={data} width={800} height={400}>
          <CartesianGrid strokeDasharray="3 3" vertical={false} />
          <XAxis dataKey="date" />
          <YAxis />
          <Tooltip />
          <Legend />
          <Bar dataKey="amount" fill="#8884d8" />
        </BarChart>
      ) : (
        <Empty />
      )}
    </Card>
  );
}

BarChartView.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.string,
      amount: PropTypes.number,
      count: PropTypes.number,
    })
  ).isRequired,
};

export default BarChartView;
