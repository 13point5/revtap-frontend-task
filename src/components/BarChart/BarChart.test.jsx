import React from "react";
import { render } from "@testing-library/react";

import BarChart from "./BarChart";

it("renders correctly when there is no data", () => {
  const { queryByTestId } = render(<BarChart data={[]} />);

  expect(queryByTestId("barchart-empty")).toBeTruthy();
});

it("renders correctly when there are few data points", () => {
  const data = [
    {
      date: "2020-09-06",
      amount: 60,
      count: 5,
    },
    {
      date: "2020-09-09",
      amount: 35,
      count: 12,
    },
    {
      date: "2020-09-012",
      amount: 72,
      count: 92,
    },
    {
      date: "2020-09-20",
      amount: 125,
      count: 45,
    },
  ];

  const component = render(<BarChart data={data} />);
  expect(component).toMatchSnapshot();
});
