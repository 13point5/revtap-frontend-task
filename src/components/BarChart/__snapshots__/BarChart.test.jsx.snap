// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are few data points 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Orders Count
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="recharts-wrapper"
            style="position: relative; cursor: default; width: 800px; height: 400px;"
          >
            <svg
              class="recharts-surface"
              height="400"
              version="1.1"
              viewBox="0 0 800 400"
              width="800"
            >
              <defs>
                <clippath
                  id="recharts2-clip"
                >
                  <rect
                    height="360"
                    width="730"
                    x="65"
                    y="5"
                  />
                </clippath>
              </defs>
              <g
                class="recharts-cartesian-grid"
              >
                <g
                  class="recharts-cartesian-grid-horizontal"
                >
                  <line
                    fill="none"
                    height="360"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="730"
                    x="65"
                    x1="65"
                    x2="795"
                    y="5"
                    y1="365"
                    y2="365"
                  />
                  <line
                    fill="none"
                    height="360"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="730"
                    x="65"
                    x1="65"
                    x2="795"
                    y="5"
                    y1="275"
                    y2="275"
                  />
                  <line
                    fill="none"
                    height="360"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="730"
                    x="65"
                    x1="65"
                    x2="795"
                    y="5"
                    y1="185"
                    y2="185"
                  />
                  <line
                    fill="none"
                    height="360"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="730"
                    x="65"
                    x1="65"
                    x2="795"
                    y="5"
                    y1="95"
                    y2="95"
                  />
                  <line
                    fill="none"
                    height="360"
                    stroke="#ccc"
                    stroke-dasharray="3 3"
                    width="730"
                    x="65"
                    x1="65"
                    x2="795"
                    y="5"
                    y1="5"
                    y2="5"
                  />
                </g>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
              >
                <line
                  class="recharts-cartesian-axis-line"
                  fill="none"
                  height="30"
                  stroke="#666"
                  width="730"
                  x="65"
                  x1="65"
                  x2="795"
                  y="365"
                  y1="365"
                  y2="365"
                />
                <g
                  class="recharts-cartesian-axis-ticks"
                >
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      stroke="#666"
                      width="730"
                      x="65"
                      x1="156.25"
                      x2="156.25"
                      y="365"
                      y1="371"
                      y2="365"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      stroke="none"
                      text-anchor="middle"
                      width="730"
                      x="156.25"
                      y="373"
                    >
                      <tspan
                        dy="0.71em"
                        x="156.25"
                      >
                        2020-09-06
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      stroke="#666"
                      width="730"
                      x="65"
                      x1="338.75"
                      x2="338.75"
                      y="365"
                      y1="371"
                      y2="365"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      stroke="none"
                      text-anchor="middle"
                      width="730"
                      x="338.75"
                      y="373"
                    >
                      <tspan
                        dy="0.71em"
                        x="338.75"
                      >
                        2020-09-09
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      stroke="#666"
                      width="730"
                      x="65"
                      x1="521.25"
                      x2="521.25"
                      y="365"
                      y1="371"
                      y2="365"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      stroke="none"
                      text-anchor="middle"
                      width="730"
                      x="521.25"
                      y="373"
                    >
                      <tspan
                        dy="0.71em"
                        x="521.25"
                      >
                        2020-09-012
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="30"
                      stroke="#666"
                      width="730"
                      x="65"
                      x1="703.75"
                      x2="703.75"
                      y="365"
                      y1="371"
                      y2="365"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="30"
                      stroke="none"
                      text-anchor="middle"
                      width="730"
                      x="703.75"
                      y="373"
                    >
                      <tspan
                        dy="0.71em"
                        x="703.75"
                      >
                        2020-09-20
                      </tspan>
                    </text>
                  </g>
                </g>
              </g>
              <g
                class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
              >
                <line
                  class="recharts-cartesian-axis-line"
                  fill="none"
                  height="360"
                  stroke="#666"
                  width="60"
                  x="5"
                  x1="65"
                  x2="65"
                  y="5"
                  y1="5"
                  y2="365"
                />
                <g
                  class="recharts-cartesian-axis-ticks"
                >
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="360"
                      stroke="#666"
                      width="60"
                      x="5"
                      x1="59"
                      x2="65"
                      y="5"
                      y1="365"
                      y2="365"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="360"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="57"
                      y="365"
                    >
                      <tspan
                        dy="0.355em"
                        x="57"
                      >
                        0
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="360"
                      stroke="#666"
                      width="60"
                      x="5"
                      x1="59"
                      x2="65"
                      y="5"
                      y1="275"
                      y2="275"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="360"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="57"
                      y="275"
                    >
                      <tspan
                        dy="0.355em"
                        x="57"
                      >
                        25
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="360"
                      stroke="#666"
                      width="60"
                      x="5"
                      x1="59"
                      x2="65"
                      y="5"
                      y1="185"
                      y2="185"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="360"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="57"
                      y="185"
                    >
                      <tspan
                        dy="0.355em"
                        x="57"
                      >
                        50
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="360"
                      stroke="#666"
                      width="60"
                      x="5"
                      x1="59"
                      x2="65"
                      y="5"
                      y1="95"
                      y2="95"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="360"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="57"
                      y="95"
                    >
                      <tspan
                        dy="0.355em"
                        x="57"
                      >
                        75
                      </tspan>
                    </text>
                  </g>
                  <g
                    class="recharts-layer recharts-cartesian-axis-tick"
                  >
                    <line
                      class="recharts-cartesian-axis-tick-line"
                      fill="none"
                      height="360"
                      stroke="#666"
                      width="60"
                      x="5"
                      x1="59"
                      x2="65"
                      y="5"
                      y1="5"
                      y2="5"
                    />
                    <text
                      class="recharts-text recharts-cartesian-axis-tick-value"
                      fill="#666"
                      height="360"
                      stroke="none"
                      text-anchor="end"
                      width="60"
                      x="57"
                      y="5"
                    >
                      <tspan
                        dy="0.355em"
                        x="57"
                      >
                        100
                      </tspan>
                    </text>
                  </g>
                </g>
              </g>
              <g
                class="recharts-layer recharts-bar"
              >
                <g
                  class="recharts-layer recharts-bar-rectangles"
                >
                  <g
                    class="recharts-layer"
                  >
                    <g
                      class="recharts-layer recharts-bar-rectangle"
                    />
                    <g
                      class="recharts-layer recharts-bar-rectangle"
                    />
                    <g
                      class="recharts-layer recharts-bar-rectangle"
                    />
                    <g
                      class="recharts-layer recharts-bar-rectangle"
                    />
                  </g>
                </g>
              </g>
            </svg>
            <div
              class="recharts-legend-wrapper"
              style="position: absolute; width: 790px; height: auto; left: 5px; bottom: 5px;"
            >
              <ul
                class="recharts-default-legend"
                style="padding: 0px; margin: 0px; text-align: center;"
              >
                <li
                  class="recharts-legend-item legend-item-0"
                  style="display: inline-block; margin-right: 10px;"
                >
                  <svg
                    class="recharts-surface"
                    height="14"
                    style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                    version="1.1"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      class="recharts-legend-icon"
                      d="M0,4h32v24h-32z"
                      fill="#8884d8"
                      stroke="none"
                    />
                  </svg>
                  <span
                    class="recharts-legend-item-text"
                  >
                    count
                  </span>
                </li>
              </ul>
            </div>
            <div
              class="recharts-tooltip-wrapper"
              style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
            >
              <div
                class="recharts-default-tooltip"
                style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
              >
                <p
                  class="recharts-tooltip-label"
                  style="margin: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      id="recharts_measurement_span"
      style="position: absolute; top: -20000px; left: 0px; padding: 0px; margin: 0px; white-space: pre;"
    >
       
    </span>
  </body>,
  "container": <div>
    <div
      class="ant-card ant-card-bordered"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Orders Count
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="recharts-wrapper"
          style="position: relative; cursor: default; width: 800px; height: 400px;"
        >
          <svg
            class="recharts-surface"
            height="400"
            version="1.1"
            viewBox="0 0 800 400"
            width="800"
          >
            <defs>
              <clippath
                id="recharts2-clip"
              >
                <rect
                  height="360"
                  width="730"
                  x="65"
                  y="5"
                />
              </clippath>
            </defs>
            <g
              class="recharts-cartesian-grid"
            >
              <g
                class="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height="360"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="730"
                  x="65"
                  x1="65"
                  x2="795"
                  y="5"
                  y1="365"
                  y2="365"
                />
                <line
                  fill="none"
                  height="360"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="730"
                  x="65"
                  x1="65"
                  x2="795"
                  y="5"
                  y1="275"
                  y2="275"
                />
                <line
                  fill="none"
                  height="360"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="730"
                  x="65"
                  x1="65"
                  x2="795"
                  y="5"
                  y1="185"
                  y2="185"
                />
                <line
                  fill="none"
                  height="360"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="730"
                  x="65"
                  x1="65"
                  x2="795"
                  y="5"
                  y1="95"
                  y2="95"
                />
                <line
                  fill="none"
                  height="360"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="730"
                  x="65"
                  x1="65"
                  x2="795"
                  y="5"
                  y1="5"
                  y2="5"
                />
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="30"
                stroke="#666"
                width="730"
                x="65"
                x1="65"
                x2="795"
                y="365"
                y1="365"
                y2="365"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    stroke="#666"
                    width="730"
                    x="65"
                    x1="156.25"
                    x2="156.25"
                    y="365"
                    y1="371"
                    y2="365"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    stroke="none"
                    text-anchor="middle"
                    width="730"
                    x="156.25"
                    y="373"
                  >
                    <tspan
                      dy="0.71em"
                      x="156.25"
                    >
                      2020-09-06
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    stroke="#666"
                    width="730"
                    x="65"
                    x1="338.75"
                    x2="338.75"
                    y="365"
                    y1="371"
                    y2="365"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    stroke="none"
                    text-anchor="middle"
                    width="730"
                    x="338.75"
                    y="373"
                  >
                    <tspan
                      dy="0.71em"
                      x="338.75"
                    >
                      2020-09-09
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    stroke="#666"
                    width="730"
                    x="65"
                    x1="521.25"
                    x2="521.25"
                    y="365"
                    y1="371"
                    y2="365"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    stroke="none"
                    text-anchor="middle"
                    width="730"
                    x="521.25"
                    y="373"
                  >
                    <tspan
                      dy="0.71em"
                      x="521.25"
                    >
                      2020-09-012
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    stroke="#666"
                    width="730"
                    x="65"
                    x1="703.75"
                    x2="703.75"
                    y="365"
                    y1="371"
                    y2="365"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    stroke="none"
                    text-anchor="middle"
                    width="730"
                    x="703.75"
                    y="373"
                  >
                    <tspan
                      dy="0.71em"
                      x="703.75"
                    >
                      2020-09-20
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="360"
                stroke="#666"
                width="60"
                x="5"
                x1="65"
                x2="65"
                y="5"
                y1="5"
                y2="365"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="360"
                    stroke="#666"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="365"
                    y2="365"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="360"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="57"
                    y="365"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      0
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="360"
                    stroke="#666"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="275"
                    y2="275"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="360"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="57"
                    y="275"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      25
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="360"
                    stroke="#666"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="185"
                    y2="185"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="360"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="57"
                    y="185"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      50
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="360"
                    stroke="#666"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="95"
                    y2="95"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="360"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="57"
                    y="95"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      75
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="360"
                    stroke="#666"
                    width="60"
                    x="5"
                    x1="59"
                    x2="65"
                    y="5"
                    y1="5"
                    y2="5"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="360"
                    stroke="none"
                    text-anchor="end"
                    width="60"
                    x="57"
                    y="5"
                  >
                    <tspan
                      dy="0.355em"
                      x="57"
                    >
                      100
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-bar"
            >
              <g
                class="recharts-layer recharts-bar-rectangles"
              >
                <g
                  class="recharts-layer"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                  <g
                    class="recharts-layer recharts-bar-rectangle"
                  />
                </g>
              </g>
            </g>
          </svg>
          <div
            class="recharts-legend-wrapper"
            style="position: absolute; width: 790px; height: auto; left: 5px; bottom: 5px;"
          >
            <ul
              class="recharts-default-legend"
              style="padding: 0px; margin: 0px; text-align: center;"
            >
              <li
                class="recharts-legend-item legend-item-0"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,4h32v24h-32z"
                    fill="#8884d8"
                    stroke="none"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                >
                  count
                </span>
              </li>
            </ul>
          </div>
          <div
            class="recharts-tooltip-wrapper"
            style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
          >
            <div
              class="recharts-default-tooltip"
              style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
            >
              <p
                class="recharts-tooltip-label"
                style="margin: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
